src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -643,12 +654,20 @@ namespace levin
         const auto start = std::chrono::steady_clock::now();
         noise_channel& channel = zone_->channels.at(channel_);
 
+        bool can_broadcast = true;
+        zone_->p2p->for_connection(channel.connection, [&](detail::p2p_context& context){
+            can_broadcast = context.can_broadcast_txs_to_peer;
+            return true;
+        });
+
         if (!channel.connection.is_nil())
         {
           epee::byte_slice message = nullptr;
-          if (!channel.active.empty())
+          if (can_broadcast && !channel.active.empty())
+          {
             message = channel.active.take_slice(zone_->noise.size());
-          else if (!channel.queue.empty())
+          }
+          else if (can_broadcast && !channel.queue.empty())
```

Comment:
This check should be reverted - a message already started on the channel, so it should continue.

Reply:
can_broadcast won't be set to false until the message is sent.

Reply:
Then why have the check here? There's probably a way to re-organize this logic since its only set at the end.